src/Random.cpp

```diff
@@ -482,15 +591,18 @@ namespace galsim {
         return oss.str();
     }
 
-    void PoissonDeviate::generateFromExpectation(int N, double* data)
+    void PoissonDeviate::generateFromExpectation(long long N, double* data)
     {
-        for (int i=0; i<N; ++i) {
+        double old_mean = getMean();
+        // Note: cannot parallelize this, since Poisson doesn't have a reliable discard.
+        for (long long i=0; i<N; ++i) {
             double mean = data[i];
             if (mean > 0.) {
                 setMean(mean);
                 data[i] = (*this)();
             }
         }
+        setMean(old_mean);
```

Comment:
Yes.  Probably not one that bit anyone, but if you did `pd.generate_from_expecation(..)` then the `pd` object ended up with the mean set to whatever the last value in that image was, rather than the value you set when first building `pd`.  There's now a unit test for it, which failed prior to this change.